### PR TITLE
Circular service dependencies and deprecated method

### PR DESCRIPTION
**Pull request by bjaxelsen**
When enabling the locale module and having active configuration overrides, we run into circular dependencies regarding the language manager service.

Steps to reproduce:

1) enable locale module and a non-English language
2) enable domain_config
3) put a domain-specific override into configuration (I used system.site)
4) Symfony service loader will complain about circular depencency on "language_manager" service

I am not sure what is the best way to avoid this problem but I tried to get around it by postponing the initialization of the domain negotiator and language manager services. Please have a look at it.

Also, entityManager is deprecated, see:
https://www.drupal.org/node/2549139

### CODE REVIEW

**Comment by agentrickard** [comment]
Can you cross-reference this with #107 and the problem I was having in https://github.com/agentrickard/domain/tree/107-optional

Does this fix help that move forward?

See also #144 and #145 regarding the entityManager. If this fixes any of those issues, please update and corss-reference with CHANGELOG.md.

File: domain/src/DomainLoader.php [hunk 1]
```diff
@@ -44,7 +44,7 @@ public function loadSchema() {
    * {@inheritdoc}
    */
   public function load($id, $reset = FALSE) {
-    $controller = \Drupal::entityManager()->getStorage('domain');
+    $controller = \Drupal::entityTypeManager()->getStorage('domain');
```
Comment (agentrickard) [inline domain/src/DomainLoader.php:48]: Shouldn't we be injecting this service?

Comment (bjaxelsen) [inline domain/src/DomainLoader.php:48]: I don't think this is possible, see #144 (but do try and prove me wrong)

**Comment by bjaxelsen** [comment]
I think this fixes #107.

**Comment by agentrickard** [comment]
I'm on a strategic retreat through Thursday, but plan on reviewing this thoroughly on Friday.

**Comment by agentrickard** [comment]
I'm going to merge this and then make some changes.

**Pull request merged**